dateutil/parser/_parser.py

```diff
@@ -1464,6 +1494,7 @@ def parse(self, tzstr):
         except (IndexError, ValueError, AssertionError):
             return None
 
+        res.unused_tokens = not {token for token,is_used in zip(l,used_tokens) if not is_used}.issubset({",",":"})
```

Comment:
This took some staring out to figure out.  If nothing else, please break this into a couple of lines starting with `unused_tokens = {token for token,is_used in zip(l,used_tokens) if not is_used}`.

Related, the name `res.unused_tokens` suggest that it is listlike giving the tokens themselves.  If this is a bool, maybe something like `res.any_unused_tokens`?

More broadly, I'd suggest the following:

Implement `used_tokens` as `used_idxs` mirroring `skipped_idxs` in `_parser._parse`.   Then each occurrence of `used_tokens[i] = True` becomes `used_idxs.append(i)`.  Then `{token for token,is_used in zip(l,used_tokens) if not is_used}` becomes:

```
unused_idxs = set(range(len_l)).difference(used_idxs)
unused_tokens = {l[n] for n in unused_idxs}
```